src/Compilers/CSharp/Portable/Symbols/Tuples/TupleFieldSymbol.cs

```diff
@@ -24,6 +24,9 @@ internal class TupleFieldSymbol : WrappedFieldSymbol
         public TupleFieldSymbol(TupleTypeSymbol container, FieldSymbol underlyingField, int tupleFieldId)
             : base(underlyingField)
         {
+            Debug.Assert(container.UnderlyingNamedType.Equals(underlyingField.ContainingType, TypeCompareKind.IgnoreDynamicAndTupleNames) || this is TupleVirtualElementFieldSymbol,
```

Comment:
The purpose of this assert is to actually check the internal implementation. I think it is ok to be strict about types here.